tests/dialects/test_tsql.py

```diff
@@ -389,7 +389,62 @@ def test_types_bin(self):
             },
         )
 
+    def test_ddl(self):
+        sql = """CREATE TABLE #mytemp (a INTEGER,  b CHAR(2), c TIME(4), d FLOAT(24));"""
+        self.validate_all(
+            " ".join(sql.split()),
+            write={
+                "tsql": "CREATE TABLE #mytemp (a INTEGER, b CHAR(2), c TIMESTAMP(4), d FLOAT(24))"
+            },
+        )
+
+    def test_transaction(self):
+        # BEGIN { TRAN | TRANSACTION }
+        #    [ { transaction_name | @tran_name_variable }
+        #    [ WITH MARK [ 'description' ] ]
+        #    ]
+        # [ ; ]
+        self.validate_identity("BEGIN TRANSACTION")
+        # self.validate_identity("BEGIN TRAN")
+        self.validate_identity("BEGIN TRANSACTION transaction_name")
+        self.validate_identity("BEGIN TRANSACTION @tran_name_variable")
+        self.validate_identity("BEGIN TRANSACTION transaction_name WITH MARK 'description'")
+
+    def test_commit(self):
+        # COMMIT [ { TRAN | TRANSACTION }  [ transaction_name | @tran_name_variable ] ] [ WITH ( DELAYED_DURABILITY = { OFF | ON } ) ] [ ; ]
+
+        self.validate_identity("COMMIT")
+        self.validate_all("COMMIT TRAN")
```

Comment:
This doesn't seem correct as well. Assuming we make the change I suggested in the parser, this should become:

```python
self.validate_all("COMMIT TRAN", write={"tsql": "COMMIT TRANSACTION"})
```

Reply:
Done.